### PR TITLE
[BUGFIX] Datasource not unregist when directory or file deleted

### PR DESCRIPTION
**Pull request by gseok**
- If dataSource is not used, then dataSource must be unregistered

Change-Id: I91e4e19ecf62e03e1395bb27a8e5b8d2bc69c239
Signed-off-by: GyeongSeok Seo <gyseok.seo@samsung.com>

### CODE REVIEW

**Comment by hwshim** [comment]
@gseok 
That's not my intention.
DataSourceRegistry exists to cache DataSources, and should be used again until it is alive in the workspace. But instead in this case, you intent delete project, 'delete project' topic or other like some message should be known by partModelProvider.js. Anyway I'll handle this. Please wait for the fix. Thanks.

**Comment by wcho** [comment]
@hwshim Please open an issue if this needs to be fixed

File: common/src/webida/plugins/workbench/ui/partModelProvider.js [hunk 1]
```diff
@@ -207,9 +207,7 @@ define([
                 var model = part.getModel();
                 if (provider.isModelProvided(model) === true
```
Comment (wcho) [inline common/src/webida/plugins/workbench/ui/partModelProvider.js]: It looks weird checking equality with boolean values in a condition expression.

**Comment by cimfalab** [comment]
@hwshim Dear,
This issue is not a problem in most normal cases.
But in some cases like repetitive integration testing, it can be annoying issue.
So @gseok raised an issue (https://github.com/webida/webida-client/issues/413) months ago.

Could you please fix this in this week? :)

**Comment by cimfalab** [comment]
@hwshim How's it going?

**Comment by hwshim** [comment]
@cimfalab 
I'll check this in this week.
#654

**Pull request closed without merging**